containers/hawkular-metrics-openshift-integration/src/main/java/org/hawkular/openshift/auth/OpenShiftTokenAuthentication.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.hawkular.openshift.auth;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URL;
+import javax.net.ssl.HttpsURLConnection;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.JsonNodeFactory;
+import org.codehaus.jackson.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author mwringe
+ */
+public class OpenShiftTokenAuthentication {
+
+    private static final Logger logger = LoggerFactory.getLogger(OpenShiftTokenAuthentication.class);
+
+    private static final String HAWKULAR_TENANT = "hawkular-tenant";
+
+    private static final String KUBERNETES_MASTER_URL =
+            System.getProperty("KUBERNETES_MASTER_URL", "https://kubernetes.default.svc.cluster.local");
+
+    //The resource to check against for security purposes. For this version we are allowing Metrics based on a users
+    //access to the pods in in a particular project.
+    private static final String RESOURCE = "pods";
+
+    private static final String KIND = "SubjectAccessReview";
+
+    enum HTTP_METHOD {GET, PUT, POST, DELETE, PATCH};
+
+    public void doFilter(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws
+            IOException, ServletException {
+
+        String token = request.getHeader(OpenShiftAuthenticationFilter.AUTHORIZATION_HEADER);
+        String tenant = request.getHeader(HAWKULAR_TENANT);
+
+        if (token == null || tenant == null) {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST,
+                               "The '" + OpenShiftAuthenticationFilter.AUTHORIZATION_HEADER + "' and '"
+                                       + HAWKULAR_TENANT + "' headers" + " are required");
+            return;
+        }
+
+        if (isAuthorized(request.getMethod(), token, tenant)) {
+            filterChain.doFilter(request, response);
+            return;
+        } else {
+            response.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
```

Comment:
OK, updated it to remove the extra returns. I like to put those returns in there to make sure that its the exit point for that method. We should always return after a sendError or doFilter, but in this case they are not needed.